components/media-player/PositionSlider.vue

```diff
@@ -0,0 +1,90 @@
+<script setup lang="ts">
+import * as slider from "@zag-js/slider";
+import { normalizeProps, useMachine } from "@zag-js/vue";
+
+const { currentPosition, currentTrackDuration } = useNuxtApp().$mediaPlayer;
+
+const newPosition = ref<number | null>(null);
+const [positionState, positionSend] = useMachine(
+  slider.machine({
+    id: "position",
+    value: [0],
+    step: 0.01,
+    onValueChange(details) {
+      const [value] = details.value;
+      if (value) newPosition.value = value;
+    },
+    onValueChangeEnd(details) {
+      const [value] = details.value;
+      if (value) {
```

Comment:
why the if (value)? is it nullable?
0 is falsy, so maybe you want to do `value != null` (null or undefined) instead?

Reply:
zag-js defines the value as number | undefined